### DISCUSSION TITLE
.NET compression

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#2510] [Questions]
I was reading about .net compression and wondering if CSLA can use this in the future?  Is it necessary to compress and decompress a object graph as it is serialized and de-serialized ?

**Comment by rockfordlhotka** [comment]
Yes, you can add compression into the data portal pipeline if you want. I think there's still a sample for this, and it is discussed in the _Using CSLA 4_ book, specifically the data portal configuration book.

You may or may not get much value out of the compression, so you should test and confirm. The data stream generated by MobileFormatter is already optimized in numerous ways, and sometimes your web server will be using http compression, so adding even _more_ compression is just a waste of CPU.

If you aren't using http compression, and if your actual data (string data mostly) can be compressed, then it can be beneficial. For example, if your object graphs often have string values containing a lot of text - like documents, etc. - then compression might be useful to shrink those big text blobs.
